library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -122,8 +130,10 @@ private static boolean shouldAbortStartingBatch(ConnectionChecker connectionChec
     private static void deleteBatchIfNeeded(InternalDownloadBatchStatus downloadBatchStatus,
                                             DownloadsBatchPersistence downloadsBatchPersistence,
                                             DownloadBatchStatusCallback callback) {
-        if (downloadBatchStatus.status() == DELETING && downloadsBatchPersistence.deleteSync(downloadBatchStatus)) {
+        if (downloadBatchStatus.status() == DELETING) {
+            Log.v("sync delete and mark as deleted batch " + downloadBatchStatus.getDownloadBatchId().rawId());
             downloadBatchStatus.markAsDeleted();
+            downloadsBatchPersistence.deleteSync(downloadBatchStatus);
```

Comment:
we need to mark the batch immediately as deleting, this is why this check is now in this position instead of being part of the `if` check. For the callback, it does not matter the result of the deletion as it has marked as deleted anyway and that is what it should be propagated.

Reply:
It doesn't matter? What happens if the files aren't deleted?